src/VisualStudio/Core/Def/Implementation/AnalyzerDependency/AnalyzerFileWatcherService.cs

```diff
@@ -65,23 +74,11 @@ internal void RemoveAnalyzerAlreadyLoadedDiagnostics(ProjectId projectId, string
             _updateSource.ClearDiagnosticsForProject(projectId, Tuple.Create(s_analyzerChangedErrorId, analyzerPath));
         }
 
-        private void RaiseAnalyzerChangedWarning(ProjectId projectId, string analyzerPath)
+        private async void RaiseAnalyzerChangedWarningAsync(ProjectId projectId, string analyzerPath)
         {
-            string message = string.Format(ServicesVSResources.WRN_AnalyzerChangedMessage, analyzerPath);
-
-            DiagnosticData data = new DiagnosticData(
-                IDEDiagnosticIds.AnalyzerChangedId,
-                FeaturesResources.ErrorCategory,
-                message,
-                ServicesVSResources.WRN_AnalyzerChangedMessage,
-                severity: DiagnosticSeverity.Warning,
-                isEnabledByDefault: true,
-                warningLevel: 0,
-                workspace: _workspace,
-                projectId: projectId,
-                title: ServicesVSResources.WRN_AnalyzerChangedTitle);
-
-            _updateSource.UpdateDiagnosticsForProject(projectId, Tuple.Create(s_analyzerChangedErrorId, analyzerPath), SpecializedCollections.SingletonEnumerable(data));
+            var messageArguments = new string[] { analyzerPath };
+            var diagnostic = await DiagnosticData.CreateAsync(_analyzerChangedRule, messageArguments, projectId, _workspace, CancellationToken.None).ConfigureAwait(false);
+            _updateSource.UpdateDiagnosticsForProject(projectId, Tuple.Create(s_analyzerChangedErrorId, analyzerPath), SpecializedCollections.SingletonEnumerable(diagnostic));
```

Comment:
is it okay to be async now? and order of event become changed? for example, it raised event (but not registered to update source yet) but before that happens, solution is closed and then it is registered to update source.